master/internal/db/postgres_users.go

```diff
@@ -425,3 +427,32 @@ func (db *PgDB) AuthTokenKeypair() (*model.AuthTokenKeypair, error) {
 		return &tokenKeypair, nil
 	}
 }
+
+// UpdateUserSetting updates user setting.
+func UpdateUserSetting(setting *model.UserWebSetting) error {
+	if len(setting.Value) == 0 {
+		_, err := Bun().NewDelete().Model(setting).Where(
+			"user_id = ?", setting.UserID).Where(
+			"storage_path = ?", setting.StoragePath).Where(
+			"key = ?", setting.Key).Exec(context.TODO())
+		return err
+	}
+
+	_, err := Bun().NewInsert().Model(setting).On("CONFLICT (user_id, key, storage_path) DO UPDATE").
```

Comment:
question:  i would have thought if we were doing `ON CONFLICT UPDATE` here, then we wouldn't need to do the deletion above. do we need both?

Reply:
Undefined `Value` in body corresponds to `Storage.remove`